community/cypher/cypher-compiler-3.1/src/main/scala/org/neo4j/cypher/internal/compiler/v3_1/planner/logical/QueryPlannerConfiguration.scala

```diff
@@ -26,6 +26,38 @@ import org.neo4j.cypher.internal.compiler.v3_1.planner.logical.steps.solveOption
 import org.neo4j.cypher.internal.compiler.v3_1.{UpdateStrategy, defaultUpdateStrategy}
 
 object QueryPlannerConfiguration {
+
+  private val leafPlanFromExpressions = Seq(
+    // MATCH (n) WHERE id(n) IN ... RETURN n
+    idSeekLeafPlanner,
+
+    // MATCH (n) WHERE n.prop IN ... RETURN n
+    uniqueIndexSeekLeafPlanner,
+
+    // MATCH (n) WHERE n.prop IN ... RETURN n
+    indexSeekLeafPlanner,
+
+    // MATCH (n) WHERE has(n.prop) RETURN n
```

Comment:
I realise you didn't author it, but this comment actually quotes a language construct that doesn't exist anymore; `has()` is now called `exists()`.